### PR TITLE
Add CVE-2024-42450

### PR DESCRIPTION
**Pull request by PiranhaSa**
The exploit leverages the following conditions in Versa Director installations:

Database Exposure: The PostgreSQL database is configured to listen on all network interfaces and uses a default, hardcoded password across all instances.
Attack Vector: Unauthenticated attackers can exploit this configuration to access the database, read local filesystem contents, and escalate privileges.

Key elements in the script:

Database Connection Check:
    The _check_db_connection function attempts to establish a connection to the database using the default credentials.
Accept Criteria:
    The accept method verifies that the target server responds to /versa/login with the expected login page contents (Versa Director).

### CODE REVIEW

**Review by ybadaoui-ostorlab — COMMENTED** [review]
Please make sure to add a short description and fix the linter!

**Review by adnaneserrar — APPROVED** [review]

File: agent/exploits/cve_2024_42450.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            host=host,
+            dbname=dbname,
+            user=user,
+            password=password,
+            connect_timeout=DEFAULT_TIMEOUT.seconds,
+        )
+        # Check the connection status
+        if connection is not None:
+            return True
+    except psycopg.OperationalError as e:
```
Comment (benyissa) [inline agent/exploits/cve_2024_42450.py:56]:
```suggestion
    except psycopg.OperationalError:
```
e is not used in the catch block

**Review by nmasdoufi-ol — APPROVED** [review]

File: agent/exploits/cve_2024_42450.py (hunk 1 above)
Comment (PiranhaSa) [reply agent/exploits/cve_2024_42450.py]: I saw it in ruff error

File: agent/exploits/cve_2024_42450.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            host=host,
+            dbname=dbname,
+            user=user,
+            password=password,
+            connect_timeout=DEFAULT_TIMEOUT.seconds,
+        )
+        # Check the connection status
+        if connection is not None:
+            return True
+    except psycopg.OperationalError:
```
[the remark this answers is not in the document]
Comment (PiranhaSa) [reply agent/exploits/cve_2024_42450.py:56]: I don't think so  , when the creds are not valid or other reason the connection is refused so the check is false and they are not vulnerable

**Review by 3asm — APPROVED** [review]

**Pull request merged**